src/main/java/com/ldtteam/structurize/api/TagManager.java

```diff
@@ -0,0 +1,79 @@
+package com.ldtteam.structurize.api;
+
+import com.ldtteam.structurize.blocks.interfaces.IAnchorBlock;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.ldtteam.structurize.api.constants.Constants.GROUNDLEVEL_TAG;
+import static com.ldtteam.structurize.api.constants.Constants.INVISIBLE_TAG;
+
+/**
+ * Handles tags.
+ */
+public class TagManager
+{
+    /**
+     * List of tag options. Mods can just insert on this in mod constructor.
+     */
+    private static List<String> GLOBAL_TAG_OPTIONS = new ArrayList<>();
+
+    /**
+     * Block specific tag options.
+     */
+    private static Map<String, Predicate<IAnchorBlock>> BLOCK_SPECIFIC_TAG_OPTIONS = new HashMap<>();
+
+    static
+    {
+        GLOBAL_TAG_OPTIONS.add(GROUNDLEVEL_TAG);
+        GLOBAL_TAG_OPTIONS.add(INVISIBLE_TAG);
+    }
+    /**
+     * Register a new global tag option.
+     * @param tag the option to register.
+     */
+    public static void registerGlobalTagOption(final String tag)
+    {
+        if (!GLOBAL_TAG_OPTIONS.contains(tag))
+        {
+            GLOBAL_TAG_OPTIONS.add(tag);
+        }
+    }
+
+    /**
+     * Register a block specific tag option.
+     * @param tag the tag to register.
+     * @param predicate the predicate that has to apply.
+     */
+    public static void registerSpecificTagOption(final String tag, final Predicate<IAnchorBlock> predicate)
+    {
+        if (BLOCK_SPECIFIC_TAG_OPTIONS.containsKey(tag))
+        {
+            BLOCK_SPECIFIC_TAG_OPTIONS.put(tag, BLOCK_SPECIFIC_TAG_OPTIONS.get(tag).or(predicate));
+        }
+        else
+        {
+            BLOCK_SPECIFIC_TAG_OPTIONS.put(tag, predicate);
+        }
```

Comment:
`BLOCK_SPECIFIC_TAG_OPTIONS.compute(tag, (k, v) -> (v == null) ? predicate : v.or(predicate));`

Reply:
I know I could one line it, but this way it's a bit more obvious what it is doing.

Reply:
`compute` is more efficient though.

Reply:
This is a one time thing on startup